compiler/rustc_parse/src/parser/nonterminal.rs

```diff
@@ -47,7 +47,7 @@ impl<'a> Parser<'a> {
                 token.can_begin_expr()
                 // This exception is here for backwards compatibility.
                 && !token.is_keyword(kw::Let)
-                && (token.span.edition().at_least_rust_2024() || !token.is_keyword(kw::Const))
+                && (token.is_keyword(kw::Const) || token.span.edition().at_least_rust_2024())
```

Comment:
Is this still correct? It looks like the `!` disappeared.

Reply:
ops! I did not note that I delete the `not`, pushing the change.

Reply:
It's a little concerning that no tests failed after that change... Maybe they didn't have time to finish yet?